packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet.tsx

```diff
@@ -139,18 +143,45 @@ const useWalletCreateDialogStyle = makeStyles((theme: Theme) =>
     }),
 )
 
+type FormData = {
+    name: string
+    confirmed?: boolean
+    mnemonic?: string
+    private_password?: string
+}
+
 export function DashboardWalletCreateDialog(props: WrappedDialogProps<object>) {
+    const name = useRef('')
     const { t } = useI18N()
     const state = useState(0)
     const classes = useWalletCreateDialogStyle()
-
-    const [name, setName] = useState('')
     const [passphrase] = useState('')
-    const [mnemonic, setMnemonic] = useState('')
-    const [privKey, setPrivKey] = useState('')
-    const [confirmed, setConfirmed] = useState(false)
+
     const [showNotification, setShowNotification] = useState(false)
 
+    const schema = useMemo(() => {
+        return zod.object({
+            name: zod
+                .string()
+                .nonempty()
+                .max(
+                    WALLET_OR_PERSONA_NAME_MAX_LEN,
+                    t('input_length_exceed_prompt', {
+                        name: t('wallet_name').toLowerCase(),
+                        length: WALLET_OR_PERSONA_NAME_MAX_LEN,
+                    }),
+                ),
+            confirmed: zod.boolean().optional(),
```

Comment:
These three fields should be required.

Reply:
Ok, this is exactly what was mentioned earlier, since there are three tabs in this Dialog and each one contains a form.But we handle the submit event on the button in the dialog's footer. If u set these fields to required, and when u trigger the handleSubmit in the first tab. It will check all the fields, But the `menomonic` and `private_password` be void. So this will cause form validation to fail. This doesn't seem to fit the scenario we want here.  So I checked them in the disabled of the button.

```
disabled={
    !(state[0] === 0 && !errors.name && watch('confirmed')) &&
    !(state[0] === 1 && !errors.name && watch('mnemonic')?.length) &&
    !(state[0] === 2 && !errors.name && watch('private_password')?.length)}
```

The best implementation is that the forms in the three tabs have their own submit buttons, which handle the corresponding submit events separately